vivarium/core/emitter.py

```diff
@@ -320,9 +322,12 @@ def __init__(self, config: Dict[str, Any]) -> None:
         self.emit_limit = config.get('emit_limit', MONGO_DOCUMENT_LIMIT)
         self.embed_path = config.get('embed_path', tuple())
 
-        # create object instance of mongo client to prevent forking issues
-        self.client = MongoClient(
-            config.get('host', self.default_host))
+        # create new MongoClient per OS process
+        curr_pid = os.getpid()
+        if curr_pid not in self.client_dict:
+            self.client_dict[curr_pid] = MongoClient(
+                config.get('host', self.default_host))
+        self.client = self.client_dict[curr_pid]
```

Comment:
what happens if the `client_dict` is not provided? Can we make the `client_dict` optional?